test/Nancy.Tests/Unit/Responses/StreamResponseFixture.cs

```diff
@@ -74,22 +74,31 @@ public void Should_throw_exception_when_stream_is_non_readable()
             exception.ShouldNotBeNull();
         }
 
-        public class TestStream : Stream
+        private class TestStream : Stream
         {
+            private readonly bool canRead;
+            private readonly bool canWrite;
+
             public TestStream(bool canRead = true, bool canWrite = true)
             {
-                this.CanRead = canRead;
-                this.CanWrite = canWrite;
+                this.canRead = canRead;
+                this.canWrite = canWrite;
             }
 
-            public override bool CanRead { get; }
+            public override bool CanRead
+            {
+                get { return this.canRead; }

```

Comment:
Why?

Reply:
Whops! That's some auto-fixing by Rider. It looks like our `.DotSettings` file is targeting C# 5 features https://github.com/NancyFx/Nancy/blob/master/Nancy.sln.DotSettings#L57
In the context of this PR, it doesn't really matter as it's a test helper class, but I'll send a separate PR for the `.DotSettings` file. Time to bump our langversion to C# 7 instead of 6?

Reply:
> Time to bump our langversion to C# 7 instead of 6?

Fine by me 😄

Reply:
Or `latest` for 7.1 ?

Reply:
I'm not sure `latest` would be optimal, in case people start using features we don't have the infrastructure to build. Latest would be 7.2, BTW 😉

Reply:
Ah true, but time to go 7?

Reply:
I think 7 would be fine, or just remove the limitation (which should default to latest major)? You can even build C# 7 with Visual Studio 2015, so I think most people would be able to contribute easily enough.

Reply:
Also, because of the community editions of Visual Studio, there's not really any good arguments to not be on latest Visual Studio anymore.

Reply:
Would we risk having CI servers fail though?

Reply:
> Would we risk having CI servers fail though?

Yes, if C# 8 came out and someone submitted a PR with a C# 8-specific feature.

Reply:
https://github.com/NancyFx/Nancy/pull/2854